bpf/bpf_netdev.c

```diff
@@ -552,7 +567,12 @@ int from_netdev(struct __sk_buff *skb)
 			tunnel_endpoint = skb->cb[4];
 			skb->mark = 0;
 			bpf_clear_cb(skb);
-			return __encap_and_redirect_with_nodeid(skb, tunnel_endpoint, seclabel, TRACE_PAYLOAD_LEN);
+
+#ifdef ENCAP_IFINDEX
+			if (tunnel_endpoint)
```

Comment:
Why is the check for `tunnel_endpoint != 0` needed?

Reply:
I thought we could work with encapsulation and direct_routing both enabled. However it doesn't seem to work for me with GKE at least so could perhaps be removed until another patch fixes up direct_routing + tunnel.

Reply:
OK. I was wondering whether it changes the behavior of the tunnel only mode as we would no longer encapsulate to 0.0.0.0 which would result in a drop but pass down to the stack.

Reply:
It actually might be CI (local tests with ipsec) failed.

Reply:
It actually might be CI (local variant with ipsec tests) failed.